src/rosdistro/manifest_provider/cache.py

```diff
@@ -130,13 +130,14 @@ def __call__(self, repo):
             logger.debug('Load package XMLs for repo "%s" from cache' % repo.name)
 
         # De-duplicate with the release package XMLs. This will cause the YAML writer
-        # to use references, saving a lot of space in the cache file.
-        for key in repo_cache or {}:
-            if key[0] != '_':
-                repo_cache[key][1] = sanitize_xml(repo_cache[key][1])
```

Comment:
Problem was here since `repo_cache[key]` is now a tuple (via the `SourceRepositoryCache.add` method) and therefore immutable.